### ISSUE TITLE
Markdownlint: Confusing or contradictory rule - expects empty line 1 vs MD041 requiring H1

### ISSUE THREAD

**Issue by konard** [codacy/codacy-analysis-cli#547]
<img width="1824" height="1101" alt="Image" src="https://github.com/user-attachments/assets/7a45fe24-ac68-4d0e-b8a1-222af9a33f26" />

## Description

I'm experiencing a contradictory markdown linting issue when using Codacy's markdownlint engine on a GitHub pull request.

## The Problem

Codacy is reporting an issue on a markdown file:

- **File**: FEATURE_COMPARISON.md
- **Line**: 1
- **Error**: "Expected: {None}; Actual: # Feature Comparison"
- **Category**: Best practice

However, this contradicts standard markdown best practice:
- **MD041** (first-line-heading) requires files to start with an H1 heading
- All other markdown files in the same repository start with H1 headings and pass Codacy checks

## Expected Behavior

Either:
1. Codacy should accept H1 headings on line 1 (consistent with MD041)
2. Or provide clear documentation about why line 1 should be empty

## Repository Context

- **Repository**: https://github.com/linksplatform/Protocols.Lino
- **Pull Request**: https://github.com/linksplatform/Protocols.Lino/pull/82
- **Affected File**: FEATURE_COMPARISON.md

All other .md files in the repository (README.md, README.ru.md, etc.) start with H1 headings on line 1 and pass Codacy checks without issues.

## Request

Please clarify:
1. Is this expected behavior?
2. Why does this rule differ from standard markdownlint MD041?
3. Should all markdown files have an empty first line, or is this a false positive?

Thank you!

**Comment by konard** [comment]
## Workarounds

I've tried several approaches to resolve this issue:

### 1. Fixed .codacy.yml Configuration
Changed from `exclude_patterns` to `exclude_paths` (proper syntax per Codacy docs):
```yaml
engines:
  markdownlint:
    enabled: true
    exclude_paths:
      - 'FEATURE_COMPARISON.md'
```

### 2. Added .markdownlint.json
Disabled MD041 globally:
```json
{
  "default": true,
  "MD041": false
}
```

### 3. Prepared YAML Frontmatter Alternative
Created version with YAML frontmatter (which MD041 allows):
```markdown
---
title: Feature Comparison
---

(rest of content)
```
